pkg/image/docker/daemon_provider.go

```diff
@@ -176,17 +196,19 @@ func (p *DaemonImageProvider) pullOptions() (types.ImagePullOptions, error) {
 	return options, nil
 }
 
-func authURL(imageStr string) (string, error) {
+func authURL(imageStr string, dockerhubWorkaround bool) (string, error) {
 	ref, err := name.ParseReference(imageStr)
 	if err != nil {
 		return "", err
 	}
 
 	url := ref.Context().RegistryStr()
-	if url == "index.docker.io" {
+	if dockerhubWorkaround && url == "index.docker.io" {
 		// why do this? There is an upstream issue here: https://github.com/docker/docker-credential-helpers/blob/e595cd69465c6b0f7af2d49582b82fdeddecbf75/wincred/wincred_windows.go#L113-L127
```

Comment:
For people coming to this PR. This comment gives a lot of the context as to why this whole daemon provider works the way it does.